### PR TITLE
9/10 move copy widgets

### PR DESCRIPTION
**Pull request by isigmund**
This does add the move/copy functionality for widgets between different dashboards

### CODE REVIEW

File: web/app/dashboard/dashboard.edit.html [hunk 1]
```diff
@@ -38,8 +38,15 @@
...
-                <div class="box-header-btns pull-right">
-                    <a class="settings-btn" title="settings" ng-click="openSettings(widget)"><i class="glyphicon glyphicon-cog"></i></a>
+                <div class="btn-group box-header-btns dropdown" uib-dropdown dropdown-append-to-body>
+                  <button id="settingsMenu" type="button" class="box-header-dropdown settings-btn" uib-dropdown-toggle>
+                      <i class="glyphicon glyphicon-option-vertical"></i>
+                  </button>
+                  <ul class="dropdown-menu" style="min-width: 90px;" uib-dropdown-menu role="menu" aria-labelledby="settingsMenu">
+                    <li><a ng-click="openSettings(widget)" class="ng-binding"><i class="glyphicon glyphicon-edit"></i> Edit ...</a></li>
+                    <li><a ng-click="transferWidget(widget)" class="ng-binding"><i class="glyphicon glyphicon-duplicate"></i> Copy/move to ...</a></li>
+                    <li><a ng-click="remove(widget)" class="ng-binding"><i class="glyphicon glyphicon-trash"></i> Delete</a></li>
```
Comment (ghys) [inline web/app/dashboard/dashboard.edit.html]: Can you add a `<li class="divider" />` before the Delete menu item? I'm still a little dubious with this, especially with no confirmation. I'm sure users will happen to tap it by accident (which is actually fine as long as they reload before clicking Save or Run, but still...)

Comment (ghys) [inline web/app/dashboard/dashboard.edit.html]: (no need for another confirmation dialog box is what I'm saying)

File: web/app/dashboard/transferwidget.html [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+            <div class="form-group" ng-class="{error: _form.targetDashboard.$error && _form.submitted}">
+                <label class="control-label col-xs-3 col-lg-3 col-md-3">to dashboard</label>
+                <div class="col-lg-5 col-md-5 col-xs-5">
+                    <select ng-model="form.targetDashboard" ng-options="dashboard.id as dashboard.name for dashboard in selDashboards" class="form-control"></select>
+                </div>
+            </div>
+        </div>
+        <div class="modal-footer">
+            <a ng-click="dismiss()" class="btn btn-default" tabindex="-1"><i class="glyphicon glyphicon-remove"></i>&nbsp;Cancel</a>
+            <a ng-click="copyWidget({close: true})" class="btn btn-primary"><i class="glyphicon glyphicon-duplicate"></i>&nbsp;Copy</a>
```
Comment (ghys) [inline web/app/dashboard/transferwidget.html]: Personal preference maybe, but I think the Move and Copy buttons should be switched, it 'feels' more natural.

File: web/app/dashboard/dashboard.edit.html [hunk 3]
```diff
@@ -38,8 +38,15 @@
...
                     </div>
                 </div>
-                <div class="box-header-btns pull-right">
-                    <a class="settings-btn" title="settings" ng-click="openSettings(widget)"><i class="glyphicon glyphicon-cog"></i></a>
+                <div class="btn-group box-header-btns dropdown" uib-dropdown dropdown-append-to-body>
+                  <button id="settingsMenu" type="button" class="box-header-dropdown settings-btn" uib-dropdown-toggle>
+                      <i class="glyphicon glyphicon-option-vertical"></i>
+                  </button>
+                  <ul class="dropdown-menu" style="min-width: 90px;" uib-dropdown-menu role="menu" aria-labelledby="settingsMenu">
+                    <li><a ng-click="openSettings(widget)" class="ng-binding"><i class="glyphicon glyphicon-edit"></i> Edit ...</a></li>
```
Comment (isigmund) [inline web/app/dashboard/dashboard.edit.html]: done

File: web/app/dashboard/dashboard.edit.html [hunk 4]
```diff
@@ -38,8 +38,15 @@
...
+                    <li><a ng-click="transferWidget(widget)" class="ng-binding"><i class="glyphicon glyphicon-duplicate"></i> Copy/move to ...</a></li>
```
Comment (isigmund) [inline web/app/dashboard/dashboard.edit.html]: done

File: web/app/dashboard/dashboard.edit.html (hunk 1 above)
Comment (isigmund) [inline web/app/dashboard/dashboard.edit.html]: done

File: web/app/dashboard/dashboard.edit.html [hunk 5]
```diff
@@ -38,8 +38,15 @@
                         {{ widget.name }}
                     </div>
                 </div>
-                <div class="box-header-btns pull-right">
-                    <a class="settings-btn" title="settings" ng-click="openSettings(widget)"><i class="glyphicon glyphicon-cog"></i></a>
+                <div class="btn-group box-header-btns dropdown" uib-dropdown dropdown-append-to-body>
+                  <button id="settingsMenu" type="button" class="box-header-dropdown settings-btn" uib-dropdown-toggle>
+                      <i class="glyphicon glyphicon-option-vertical"></i>
+                  </button>
+                  <ul class="dropdown-menu" style="min-width: 90px;" uib-dropdown-menu role="menu" aria-labelledby="settingsMenu">
```
Comment (isigmund) [inline web/app/dashboard/dashboard.edit.html]: done

**Comment by isigmund** [comment]
Dont worry I am perfectly ok with your comments and remarks...
I made the suggested changes and pushed them again.

Take care
   Ingo

**Comment by ghys** [comment]
Great!
Thanks again Ingo for your work.

**Pull request merged**